service/history/workflow/mutable_state_impl.go

```diff
@@ -3365,8 +3364,8 @@ func (ms *MutableStateImpl) ReplicateTimerStartedEvent(
 	timerID := attributes.GetTimerId()
 
 	startToFireTimeout := timestamp.DurationValue(attributes.GetStartToFireTimeout())
-	// TODO: Time skew need to be taken in to account.
-	expiryTime := timestamp.TimeValue(event.GetEventTime()).Add(startToFireTimeout) // should use the event time, not now
```

Comment:
This comment didn't seem to make sense -- the code is using the current event time.

Reply:
Code is using `event.GetEventTime()` which is not now when the event is replicated from a different cluster.